src/app/static/src/app/components/plots/useUpdateScale.ts

```diff
@@ -1,18 +1,82 @@
 import {useStore} from "vuex";
 import {RootState} from "../../root";
 import {ScaleSettings} from "../../store/plotState/plotState";
+import {getIndicatorMetadata, scaleStepFromMetadata} from "./utils";
+import {computed} from "vue";
 
 export const useUpdateScale = () => {
     const store = useStore<RootState>();
 
-    const updateOutputColourScale = (indicator: string, newScaleSettings: ScaleSettings) => {
+    const getScaleStep = (name: string) => {
+        if (name === "colour") {
+            return colourScaleStep.value;
+        } else if (name === "size") {
+            return sizeScaleStep.value;
+        } else {
+            throw new RangeError(`Scale type must be one of 'name' or 'colour', got ${name}`)
+        }
+    };
```

Comment:
Kept this in as otherwise typescript was giving me a warning about no return from this. And if I return undefined or something then I end up with having to manage that type elsewhere